kiwi/bootloader/config/systemd_boot.py

```diff
@@ -0,0 +1,159 @@
+        """
+        Handle EFI images
+
+        For systemd boot EFI images are provided along with systemd.
+        Thus the only action taken care of is the creation of the
+        ESP path
+
+        :param str target:
+            target identifier, one of disk, live(iso) or install(iso)
+        """
+        self.efi_boot_path = self.create_efi_path()
+
+    def setup_loader(self, target: str) -> None:
+        """
+        Setup ESP for systemd-boot using bootctl and kernel-install
+
+        For disk images only, setup the ESP layout for systemd boot
+        using bootctl. All data will be loaded from the ESP. The
+        kernel-install script provided with systemd is used to
+        manage kernel updates and the respective loader entries.
+        Distributions integrating well with systemd runs kernel-install
+        as part of their kernel packaging. In kiwi we use kernel-install
+        for the initial boot setup
+
+        :param str target:
+            target identifier, one of disk, live(iso) or install(iso)
+            Currently only the disk identifier is supported
+        """
+        if target != self.target.disk:
+            raise KiwiBootLoaderTargetError(
+                'systemd-boot is only supported with the EFI disk image target'
+            )
+        boot_options = self.custom_args['boot_options']
+        self._mount_system(
+            boot_options.get('root_device'),
+            boot_options.get('boot_device'),
+            boot_options.get('efi_device'),
+            boot_options.get('system_volumes')
+        )
+        kernel_info = Kernel(self.root_mount.mountpoint).get_kernel()
+        if not kernel_info:
+            raise KiwiKernelLookupError(
+                'No kernel in image tree found'
+            )
+        Command.run(
+            [
+                'chroot', self.root_mount.mountpoint, 'bootctl', 'install',
+                '--esp-path=/boot/efi',
+                '--no-variables',
+                '--entry-token', 'os-id'
+            ]
+        )
+        Path.wipe(f'{self.root_mount.mountpoint}/boot/loader')
+        Command.run(
+            [
+                'chroot', self.root_mount.mountpoint,
+                'kernel-install', 'add', kernel_info.version,
+                f'/boot/{kernel_info.name}',
+                f'/boot/initramfs-{kernel_info.version}.img'
+            ]
+        )
+        BootLoaderSystemdBoot._write_config_file(
+            BootLoaderTemplateSystemdBoot().get_loader_template(),
+            self.root_mount.mountpoint + '/boot/efi/loader/loader.conf',
+            self._get_template_parameters()
+        )
+
+    def set_loader_entry(self, target: str) -> None:
+        """
+        Setup/update loader entries
+
+        loader entries for systemd-boot are expected to be managed
+        through kernel-install. However, custom kiwi settings for the
+        kernel cmdline are not taken into account. Thus this method
```

Comment:
Do you write these custom settings to /etc/cmdline? They'll be taken into account by kernel-install (dracut) if you do that.

Reply:
this should be the case for all bls-based things, so this might make sense in the base class?

Reply:
I'm sorry to say that but kernel-install doesn't take them into account correctly. kernel-install at least on Fedora 37 took the host kernel cmdline which is of course wrong. In addition kiwi supports custom kernel options via the ```kernelcmdline``` attribute and those should find its way to the config. So yes we are making sure the passed kernel cmdline is correct and expected

Reply:
@schaefi `kernel-install` should take `/etc/kernel/cmdline` into account and that should take precedence over `/proc/cmdline` if is present. That's what the `kerne-install` man page documents and if that's not the case, it's a bug.

In fact, the order should be `/etc/kenel/cmdline`, `/usr/lib/kernel/cmdline`, `/proc/cmdline`.

Reply:
Unless the distribution replaces or otherwise breaks `kernel-install`, `/etc/kernel/cmdline` should work everywhere.

Reply:
:+1: great, I will adapt to this concept and update the PR soon

Reply:
yes, what @Conan-Kudo said :)

Reply:
ok done, prior kernel-install ```etc/kernel/cmdline``` is written to the system. I can confirm this worked. Thanks